src/v2/wallet.js

```diff
@@ -1116,9 +1114,27 @@ Wallet.prototype.prebuildAndSignTransaction = function(params, callback) {
       throw error;
     }
 
-    // the prebuild can be overridden by providing an explicit tx
-    const txPrebuild = params.prebuildTx || (yield this.prebuildTransaction(params));
-    const userKeychain = yield this.baseCoin.keychains().get({ id: this._wallet.keys[0], reqId: params.reqId });
+    // call prebuildTransaction and keychains-get in parallel
+
+    // see if the user provided keychains in the params
+    const providedKeychains = params.verification && params.verification.keychains;
+
+    const response = yield Promise.props({
```

Comment:
This logic is actually coin-specific, (we don't need keychains to verify XRP transactions, for example).